src/tigr_bitmaps.c

```diff
@@ -206,15 +206,22 @@ void tigrFillRect(Tigr* bmp, int x, int y, int w, int h, TPixel color) {
 
 void tigrRect(Tigr* bmp, int x, int y, int w, int h, TPixel color) {
     int x1, y1;
-    if (w <= 0 || h <= 0)
+    if (w <= 0 || h <= 0) {
         return;
-
-    x1 = x + w - 1;
-    y1 = y + h - 1;
-    tigrLine(bmp, x, y, x1, y, color);
-    tigrLine(bmp, x1, y, x1, y1, color);
-    tigrLine(bmp, x1, y1, x, y1, color);
-    tigrLine(bmp, x, y1, x, y, color);
+    } else if (w == 1 && h != 1) {
+        tigrLine(bmp, x, y - h + 1, x, y + h - 1, color);

```

Comment:
I don't get why `y0` is set to `y - h + 1` and `y1` is set to `y + h - 1` et.c.

Wouldn't it suffice to just do like this?
```c
if (w == 1) {
    tigrLine(bmp, x, y, x, y + h, color);
} else if (h == 1) {
    tigrLine(bmp, x, y, x + w, y, color);
} else { ...
```

Reply:
Yeah probably! I didn’t spend too much time deeply thinking about it, so if you see a better way then by all means